llm/api.py

```diff
@@ -38,35 +48,112 @@ def create_chat(request):
+                            "translation_confidence",
+                        ],
+                    },
+                }
+            ],
+            function_call={"name": "detect_languages"},
+            temperature=0,
+        )
 
-        response = run_chat_chain(
-            prompt=prompt,
-            session_id=session_id,
-            primary_language=primary_language,
-            english_translation_prompt=english_translation_prompt,
-            organization_id=organization.id,
-            gpt_model=gpt_model,
+        language_results = json.loads(
+            response["choices"][0]["message"]["function_call"]["arguments"]
+        )
+        logger.info("Fetched language results via function calls")
+        logger.info("language detected", language_results["language"])
+
+        # 2. Pull relevant chunks from vector database
+        prompt_embeddings = openai.Embedding.create(
+            model="text-embedding-ada-002", input=prompt
+        )["data"][0]["embedding"]
+
+        results = Embedding.objects.alias(
+            distance=L2Distance("text_vectors", prompt_embeddings)
+        ).filter(distance__gt=0.7)
+
+        relevant_english_context = "".join(result.original_text for result in results)
+        logger.info("retrieved the relevant document context from db")
+
+        # 3. Retrievial question and answer (2nd call to OpenAI, use language from 1. to help LLM respond in same language as user question)
+        response = openai.ChatCompletion.create(
+            model=gpt_model,
+            messages=context_prompt_messages(
+                organization.id,
+                language_results["language"],
+                relevant_english_context,
+                language_results["english_translation"],
+            ),
         )
+        logger.info("received response from the ai bot for the current prompt")
 
-        print(f"Chat chain result: {response}")
+        prompt_ans = response.choices[0].message.content
 
-        del response["source_documents"]
+        # 4. Fetch the chat history from our message store
+        historical_chats = MessageStore.objects.filter(session_id=session_id).all()
+
+        # 5. Store the current question and ans to the message store
+        MessageStore.objects.create(session_id=session_id, type="human", message=prompt)
+        MessageStore.objects.create(
+            session_id=session_id, type="ai", message=prompt_ans
+        )
 
         return Response(
             {
-                "answer": response["result"],
-                "chat_history": response["chat_history"],
+                "answer": prompt_ans,
+                "chat_history": [
+                    [
+                        ["content", chat.message],
+                        ["additional_kwargs", {}],
+                        ["type", chat.type],
+                        ["example", False],
+                    ]
+                    for chat in historical_chats
```

Comment:
```suggestion
                    {
                      "role": chat.role,
                      "message": chat.message
                    } for chat in historical_chats
```

Reply:
I though we wanted the response schema as mentioned in the README file, assumed it would be to process things in glific. Or was it due to langchain's output.

Reply:
Yeah that was formatted that way by langchain but since we are focused on OpenAI, sticking with a simpler format seems best.